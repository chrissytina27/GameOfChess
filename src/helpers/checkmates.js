import King from "../components/ChessPieces/King";
import Queen from "../components/ChessPieces/Queen";
import Bishop from "../components/ChessPieces/Bishop";
import Knight from "../components/ChessPieces/Knight";
import Rook from "../components/ChessPieces/Rook";
import Pawn from "../components/ChessPieces/Pawn";

//black is checkmated
export const checkmateBlack1 = () => {
  const squares = Array(64).fill(null);
  squares[7] = { piece: new King(0), moves: [] };
  squares[9] = { piece: new Rook(1), moves: [] };
  squares[16] = { piece: new Rook(1), moves: [] };
  squares[59] = { piece: new King(1), moves: [] };
  return squares;
};

//white is checkmated
export const checkmateWhite1 = () => {
  const squares = Array(64).fill(null);
  squares[7] = { piece: new King(1), moves: [] };
  squares[9] = { piece: new Rook(0), moves: [] };
  squares[16] = { piece: new Rook(0), moves: [] };
  squares[59] = { piece: new King(0), moves: [] };
  return squares;
};

export const checkmateBlack2 = () => {
  const squares = Array(64).fill(null);
  squares[4] = { piece: new King(0), moves: [] };
  squares[12] = { piece: new Pawn(1), moves: [] };
  squares[19] = { piece: new Pawn(1), moves: [] };
  squares[20] = { piece: new King(1), moves: [] };
  return squares;
};

export const checkmateWhite2 = () => {
  const squares = Array(64).fill(null);
  squares[4] = { piece: new King(1), moves: [] };
  squares[12] = { piece: new Pawn(0), moves: [] };
  squares[19] = { piece: new Pawn(0), moves: [] };
  squares[20] = { piece: new King(0), moves: [] };
  return squares;
};

export const checkmateBlack3 = () => {
  const squares = Array(64).fill(null);
  squares[6] = { piece: new King(0), moves: [] };
  squares[13] = { piece: new Pawn(0), moves: [] };
  squares[14] = { piece: new Pawn(0), moves: [] };
  squares[15] = { piece: new Pawn(0), moves: [] };
  squares[22] = { piece: new Rook(0), moves: [] };
  squares[47] = { piece: new Pawn(1), moves: [] };
  squares[53] = { piece: new Pawn(1), moves: [] };
  squares[54] = { piece: new Pawn(1), moves: [] };
  squares[56] = { piece: new Rook(1), moves: [] };
  squares[62] = { piece: new King(1), moves: [] };
  return squares;
};

export const checkmateWhite3 = () => {
  const squares = Array(64).fill(null);
  squares[6] = { piece: new King(1), moves: [] };
  squares[13] = { piece: new Pawn(1), moves: [] };
  squares[14] = { piece: new Pawn(1), moves: [] };
  squares[15] = { piece: new Pawn(1), moves: [] };
  squares[22] = { piece: new Rook(1), moves: [] };
  squares[47] = { piece: new Pawn(0), moves: [] };
  squares[53] = { piece: new Pawn(0), moves: [] };
  squares[54] = { piece: new Pawn(0), moves: [] };
  squares[56] = { piece: new Rook(0), moves: [] };
  squares[62] = { piece: new King(0), moves: [] };
  return squares;
};

export const checkmateBlack4 = () => {
  const squares = Array(64).fill(null);
  squares[0] = { piece: new Rook(0), moves: [] };
  squares[5] = { piece: new Rook(0), moves: [] };
  squares[6] = { piece: new King(0), moves: [] };
  squares[8] = { piece: new Pawn(0), moves: [] };
  squares[9] = { piece: new Pawn(0), moves: [] };
  squares[10] = { piece: new Pawn(0), moves: [] };
  squares[13] = { piece: new Pawn(0), moves: [] };
  squares[14] = { piece: new Pawn(0), moves: [] };
  squares[15] = { piece: new Pawn(0), moves: [] };

  squares[41] = { piece: new Pawn(1), moves: [] };
  squares[48] = { piece: new Pawn(1), moves: [] };
  squares[49] = { piece: new Queen(1), moves: [] };
  squares[53] = { piece: new Pawn(1), moves: [] };
  squares[54] = { piece: new Pawn(1), moves: [] };
  squares[55] = { piece: new Pawn(1), moves: [] };
  squares[56] = { piece: new Bishop(1), moves: [] };
  squares[61] = { piece: new Rook(1), moves: [] };
  squares[62] = { piece: new King(1), moves: [] };
  return squares;
};

export const checkmateWhite4 = () => {
  const squares = Array(64).fill(null);
  squares[0] = { piece: new Rook(1), moves: [] };
  squares[5] = { piece: new Rook(1), moves: [] };
  squares[6] = { piece: new King(1), moves: [] };
  squares[8] = { piece: new Pawn(1), moves: [] };
  squares[9] = { piece: new Pawn(1), moves: [] };
  squares[10] = { piece: new Pawn(1), moves: [] };
  squares[13] = { piece: new Pawn(1), moves: [] };
  squares[14] = { piece: new Pawn(1), moves: [] };
  squares[15] = { piece: new Pawn(1), moves: [] };

  squares[41] = { piece: new Pawn(0), moves: [] };
  squares[48] = { piece: new Pawn(0), moves: [] };
  squares[49] = { piece: new Queen(0), moves: [] };
  squares[53] = { piece: new Pawn(0), moves: [] };
  squares[54] = { piece: new Pawn(0), moves: [] };
  squares[55] = { piece: new Pawn(0), moves: [] };
  squares[56] = { piece: new Bishop(0), moves: [] };
  squares[61] = { piece: new Rook(0), moves: [] };
  squares[62] = { piece: new King(0), moves: [] };
  return squares;
};

export const checkmateBlack5 = () => {
  const squares = Array(64).fill(null);
  squares[0] = { piece: new Rook(0), moves: [] };
  squares[6] = { piece: new Rook(0), moves: [] };
  squares[7] = { piece: new King(0), moves: [] };
  squares[8] = { piece: new Pawn(0), moves: [] };
  squares[9] = { piece: new Pawn(0), moves: [] };
  squares[10] = { piece: new Pawn(0), moves: [] };
  squares[14] = { piece: new Pawn(0), moves: [] };
  squares[15] = { piece: new Pawn(0), moves: [] };

  squares[28] = { piece: new Knight(1), moves: [] };
  squares[41] = { piece: new Pawn(1), moves: [] };
  squares[48] = { piece: new Pawn(1), moves: [] };
  squares[53] = { piece: new Pawn(1), moves: [] };
  squares[54] = { piece: new Pawn(1), moves: [] };
  squares[55] = { piece: new Pawn(1), moves: [] };
  squares[61] = { piece: new Rook(1), moves: [] };
  squares[62] = { piece: new King(1), moves: [] };
  return squares;
};

export const checkmateWhite5 = () => {
  const squares = Array(64).fill(null);
  squares[0] = { piece: new Rook(1), moves: [] };
  squares[6] = { piece: new Rook(1), moves: [] };
  squares[7] = { piece: new King(1), moves: [] };
  squares[8] = { piece: new Pawn(1), moves: [] };
  squares[9] = { piece: new Pawn(1), moves: [] };
  squares[10] = { piece: new Pawn(1), moves: [] };
  squares[14] = { piece: new Pawn(1), moves: [] };
  squares[15] = { piece: new Pawn(1), moves: [] };

  squares[28] = { piece: new Knight(0), moves: [] };
  squares[41] = { piece: new Pawn(0), moves: [] };
  squares[48] = { piece: new Pawn(0), moves: [] };
  squares[53] = { piece: new Pawn(0), moves: [] };
  squares[54] = { piece: new Pawn(0), moves: [] };
  squares[55] = { piece: new Pawn(0), moves: [] };
  squares[61] = { piece: new Rook(0), moves: [] };
  squares[62] = { piece: new King(0), moves: [] };
  return squares;
};
