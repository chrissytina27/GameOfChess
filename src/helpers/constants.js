export const constants = {
  squareColors: {
    light: "light",
    dark: "dark"
  },
  pieceNames: {
    bishop: "bishop",
    king: "king",
    knight: "knight",
    pawn: "pawn",
    queen: "queen",
    rook: "rook"
  }
};
