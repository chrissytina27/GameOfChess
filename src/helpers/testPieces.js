import Bishop from "../components/ChessPieces/Bishop";
import ChessPiece from "../components/ChessPieces/Bishop";
import King from "../components/ChessPieces/King";
import Knight from "../components/ChessPieces/Knight";
import Pawn from "../components/ChessPieces/Pawn";
import Queen from "../components/ChessPieces/Queen";
import Rook from "../components/ChessPieces/Rook";

export const bishopB = new Bishop(0); //black knight
export const bishopW = new Bishop(1); //white knight

const piece = { blackUrl: "", name: "cp" };
export const cp = new ChessPiece(0, piece);

export const knightB = new Knight(0); //black knight

export const kingB = new King(0); //black king
export const kingW = new King(1); //white king

export const pawnB = new Pawn(0); //black pawn
export const pawnW = new Pawn(1); //white pawn
export const pawnB2 = new Pawn(0);
export const pawnW2 = new Pawn(1);

export const queenB = new Queen(0); //black queen
export const queenW = new Queen(1); //white queen

export const rookB = new Rook(0); //black rook
export const rookW = new Rook(1); //white rook
