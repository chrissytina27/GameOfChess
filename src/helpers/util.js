import { constants as c } from "./constants";
export const util = {
  diffTeam: (piece1, piece2) => piece1.getPlayer() !== piece2.getPlayer(),
  isEven: num => num % 2 === 0,
  getRow: square => Math.floor(square / 8),
  getCol: square => square % 8,
  rowDiff: (from, to) => Math.abs(util.getRow(from) - util.getRow(to)),
  colDiff: (from, to) => Math.abs(util.getCol(from) - util.getCol(to)),
  sameRow: (from, to) => Math.floor(from / 8) === Math.floor(to / 8),
  sameCol: (from, to) => util.colDiff(from, to) === 0,
  isValid: square => square >= 0 && square < 64,
  isValidMove: toPc => (toPc && toPc.name === c.pieceNames.king ? false : true),
  goNorth: from => (from - 8 > -1 ? from - 8 : -1),
  goSouth: from => (from + 8 < 64 ? from + 8 : -1),
  goEast: from => {
    const east = from + 1;
    return util.isValid(east) && util.sameRow(from, east) ? east : -1;
  },
  goWest: from => {
    const west = from - 1;
    return util.isValid(west) && util.sameRow(from, west) ? west : -1;
  },
  goNE: from => {
    const ne = from - 7,
      rowDiff = util.rowDiff(ne, from);
    return util.isValid(ne) && rowDiff === 1 ? ne : -1;
  },
  goNW: from => {
    const nw = from - 9,
      rowDiff = util.rowDiff(nw, from);
    return util.isValid(nw) && rowDiff === 1 ? nw : -1;
  },
  goSE: from => {
    const se = from + 9,
      rowDiff = util.rowDiff(se, from);
    return util.isValid(se) && rowDiff === 1 ? se : -1;
  },
  goSW: from => {
    const sw = from + 7,
      rowDiff = util.rowDiff(sw, from);
    return util.isValid(sw) && rowDiff === 1 ? sw : -1;
  },
  LShape: (x, y, sideways) => {
    const rowDiff = sideways ? 1 : 2,
      colDiff = sideways ? 2 : 1;
    return util.rowDiff(x, y) === rowDiff && util.colDiff(x, y) === colDiff;
  },
  verticalL: (x, y) => util.LShape(x, y, false),
  horizontalL: (x, y) => util.LShape(x, y, true),
  squareName: square => {
    const row = util.getRow(square) + 1,
      col = util.getCol(square),
      dec = col + 65,
      letter = String.fromCharCode(dec);
    return letter + row;
  },
  getKingIndex: (board, currentPlayer) => {
    const kingIndex = board.findIndex(
      (square, index) =>
        square &&
        square.piece.name === c.pieceNames.king &&
        square.piece.getPlayer() === currentPlayer
    );
    return kingIndex;
  },
  clearPath: (path, fromPiece, board) => {
    return path.every((square, index) => {
      if (index === path.length - 1) {
        if (board[square] && util.diffTeam(fromPiece, board[square].piece))
          return true;
        else if (!board[square]) return true;
      } else if (!board[square]) return true;
      return false;
    });
  },
  isChecker: (square, index, king, kingIndex, currentPlayer, board) => {
    if (square && square.piece.getPlayer() !== currentPlayer) {
      const otherPlayerPiece = square.piece;
      var path =
        otherPlayerPiece.name !== c.pieceNames.pawn
          ? otherPlayerPiece.getPath(otherPlayerPiece, null, index, kingIndex)
          : otherPlayerPiece.killPath(otherPlayerPiece, king, index, kingIndex);
      if (path.length) {
        if (util.clearPath(path, otherPlayerPiece, board)) {
          const foundKing = path.findIndex(square => square === kingIndex);
          if (foundKing > -1) return true;
        }
      }
    }
    return false;
  },
  isCheck: (currentPlayer, board) => {
    var check = false;
    const kingIndex = util.getKingIndex(board, currentPlayer);
    const king = board[kingIndex].piece;
    board.every((square, index) => {
      const isChecker = util.isChecker(
        square,
        index,
        king,
        kingIndex,
        currentPlayer,
        board
      );
      if (isChecker) check = true;
      return !isChecker;
    });
    return check;
  },
  checkKings: (currentPlayer, board) => {
    const otherPlayer = currentPlayer ? 0 : 1;
    const isCurrentPlayerInCheck = util.isCheck(currentPlayer, board);
    const isOtherPlayerInCheck = util.isCheck(otherPlayer, board);
    return currentPlayer
      ? [isOtherPlayerInCheck, isCurrentPlayerInCheck]
      : [isCurrentPlayerInCheck, isOtherPlayerInCheck];
  },
  isCheckmate: (currentPlayer, board) => {
    var checkmate = true;
    var futureBoard = board.slice();
    const isCheck = util.isCheck(currentPlayer, board);
    if (!isCheck) return false;

    board.forEach((square, boardIndex) => {
      if (square && square.piece.getPlayer() === currentPlayer) {
        const piece = square.piece;
        const allMoves = piece.allMoves(boardIndex);
        allMoves.forEach((move, moveIndex) => {
          const path = piece.getPath(piece, null, boardIndex, move);
          if (path.length) {
            const clearPath = util.clearPath(path, piece, board);
            if (clearPath) {
              futureBoard = board.slice();
              futureBoard[move] = { piece };
              futureBoard[boardIndex] = null;
              const isCheck = util.isCheck(currentPlayer, futureBoard);
              if (!isCheck) checkmate = false;
            }
          }
        });
      }
    });
    return checkmate;
  },
  isStalemate: (currentPlayer, board) => {
    var stalemate = true;
    var futureBoard = board.slice();
    const isCheck = util.isCheck(currentPlayer, board);
    if (isCheck) return false;
    board.forEach((square, boardIndex) => {
      if (square && square.piece.getPlayer() === currentPlayer) {
        const piece = square.piece;
        const allMoves = piece.allMoves(boardIndex);
        allMoves.forEach((move, moveIndex) => {
          const path = piece.getPath(piece, null, boardIndex, move);
          if (path.length) {
            const clearPath = util.clearPath(path, piece, board);
            if (clearPath) {
              futureBoard = board.slice();
              futureBoard[move] = { piece };
              futureBoard[boardIndex] = null;
              const isCheck = util.isCheck(currentPlayer, futureBoard);
              if (!isCheck) stalemate = false;
            }
          }
        });
      }
    });
    return stalemate;
  }
};
