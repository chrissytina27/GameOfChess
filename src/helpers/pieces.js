import { util } from "./util";
import { constants as c } from "./constants";
export const pieces = {
  king: {
    name: c.pieceNames.king,
    whiteUrl:
      "https://upload.wikimedia.org/wikipedia/commons/4/42/Chess_klt45.svg",
    blackUrl:
      "https://upload.wikimedia.org/wikipedia/commons/f/f0/Chess_kdt45.svg",
    validMoves: [
      util.goNorth,
      util.goSouth,
      util.goEast,
      util.goWest,
      util.goNE,
      util.goNW,
      util.goSE,
      util.goSW
    ]
  },
  queen: {
    name: c.pieceNames.queen,
    whiteUrl:
      "https://upload.wikimedia.org/wikipedia/commons/1/15/Chess_qlt45.svg",
    blackUrl:
      "https://upload.wikimedia.org/wikipedia/commons/4/47/Chess_qdt45.svg"
  },
  knight: {
    name: c.pieceNames.knight,
    whiteUrl:
      "https://upload.wikimedia.org/wikipedia/commons/7/70/Chess_nlt45.svg",
    blackUrl:
      "https://upload.wikimedia.org/wikipedia/commons/e/ef/Chess_ndt45.svg",
    validMoves: [
      { steps: 15, fn: util.verticalL },
      { steps: 17, fn: util.verticalL },
      { steps: 6, fn: util.horizontalL },
      { steps: 10, fn: util.horizontalL }
    ]
  },
  rook: {
    name: c.pieceNames.rook,
    whiteUrl:
      "https://upload.wikimedia.org/wikipedia/commons/7/72/Chess_rlt45.svg",
    blackUrl:
      "https://upload.wikimedia.org/wikipedia/commons/f/ff/Chess_rdt45.svg"
  },
  pawn: {
    name: c.pieceNames.pawn,
    whiteUrl:
      "https://upload.wikimedia.org/wikipedia/commons/4/45/Chess_plt45.svg",
    blackUrl:
      "https://upload.wikimedia.org/wikipedia/commons/c/c7/Chess_pdt45.svg",
    validMoves: [[util.goNorth], [util.goSouth]],
    killDir: [[util.goNW, util.goNE], [util.goSW, util.goSE]]
  },
  bishop: {
    name: c.pieceNames.bishop,
    whiteUrl:
      "https://upload.wikimedia.org/wikipedia/commons/b/b1/Chess_blt45.svg",
    blackUrl:
      "https://upload.wikimedia.org/wikipedia/commons/9/98/Chess_bdt45.svg"
  }
};

export const whitePlayer = require("./../../public/ChessPieceIcons/white_pawn_player.png");
export const blackPlayer = require("./../../public/ChessPieceIcons/black_pawn_player.png");
