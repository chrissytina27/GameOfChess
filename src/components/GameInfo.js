import React from "react";
import PropTypes from "prop-types";
import { whitePlayer, blackPlayer } from "../helpers/pieces";
import Table from "./Table";

const GameInfo = props => {
  const renderDot = player =>
    player === props.playerTurn ? <span className="dot" /> : <span />;

  const graveyardTxt =
    ", please select a piece from the graveyard to replace your pawn.";
  const invalidMove = props.invalidMove ? "Invalid move!" : "";
  const stalemate = props.stalemate ? "Stalemate!" : "";
  const checkmate =
    props.checkmate === 1
      ? "Checkmate! Black wins!"
      : props.checkmate === 0
      ? "Checkmate! White wins!"
      : "";
  const graveyardPick = props.graveyardPick
    ? props.playerTurn
      ? `White ${graveyardTxt}`
      : `Black ${graveyardTxt}`
    : "";

  return (
    <div className="nav-wrapper box2">
      <div className="nav1">
        <div className="icon">
          <img
            className="player-icon1"
            src={whitePlayer}
            alt="White Player Logo"
          />
        </div>
      </div>
      <div className="nav2">{renderDot(1)}</div>
      <div className="nav3">
        <div className="check">{graveyardPick}</div>
        <div className="check">{invalidMove}</div>
        <div className="check">{checkmate}</div>
        <div className="check">{stalemate}</div>
        <div className="table-box">
          <Table players={props.players} />
        </div>
      </div>
      <div className="nav6">
        <div className="icon-dot">
          <img
            className="player-icon2"
            src={blackPlayer}
            alt="Black Player Logo"
          />
        </div>
      </div>
      <div className="nav7"> {renderDot(0)}</div>
      <div className="nav8">
        <button onClick={props.endGame}>End Game</button>
        <button onClick={props.resetGame}>Reset Game</button>
      </div>
    </div>
  );
};
GameInfo.propTypes = {
  playerTurn: PropTypes.number,
  graveyardPick: PropTypes.bool,
  check: PropTypes.array,
  checkmate: PropTypes.oneOfType([PropTypes.bool, PropTypes.number]),
  invalidMove: PropTypes.bool,
  staleMate: PropTypes.bool,
  endGame: PropTypes.func,
  resetGame: PropTypes.func,
  players: PropTypes.array
};
export default GameInfo;
