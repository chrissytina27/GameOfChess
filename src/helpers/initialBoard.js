import King from "../components/ChessPieces/King";
import Queen from "../components/ChessPieces/Queen";
import Bishop from "../components/ChessPieces/Bishop";
import Knight from "../components/ChessPieces/Knight";
import Rook from "../components/ChessPieces/Rook";
import Pawn from "../components/ChessPieces/Pawn";
import { pieces } from "./pieces";

const squarePiece = piece => ({ piece, moves: [] });

const emptyBoard = () => Array(64).fill(null);

export const initialBoard = () => {
  const squares = emptyBoard();
  squares[0] = squarePiece(new Rook(1));
  squares[7] = squarePiece(new Rook(1));
  squares[1] = squarePiece(new Knight(1));
  squares[6] = squarePiece(new Knight(1));
  squares[2] = squarePiece(new Bishop(1));
  squares[5] = squarePiece(new Bishop(1));
  squares[3] = squarePiece(new Queen(1));
  squares[4] = squarePiece(new King(1));
  for (let i = 8; i < 16; i++) {
    squares[i] = squarePiece(new Pawn(1));
  }
  squares[56] = squarePiece(new Rook(0));
  squares[63] = squarePiece(new Rook(0));
  squares[57] = squarePiece(new Knight(0));
  squares[62] = squarePiece(new Knight(0));
  squares[58] = squarePiece(new Bishop(0));
  squares[61] = squarePiece(new Bishop(0));
  squares[59] = squarePiece(new Queen(0));
  squares[60] = squarePiece(new King(0));
  for (let i = 48; i < 56; i++) {
    squares[i] = squarePiece(new Pawn(0));
  }
  return squares;
};

export const pawnpromoBoard = () => {
  const squares = initialBoard();
  squares[43] = squares[11];
  squares[11] = null;
  squares[20] = squares[52];
  squares[52] = null;
  return squares;
};

export const partialBoard = () => {
  const squares = initialBoard();
  squares[27] = squares[11];
  squares[36] = squares[52];
  squares[39] = squares[15];
  squares[11] = null;
  squares[15] = null;
  squares[52] = null;
  squares[59] = null;
  return squares;
};

//move queen 62 -> 44 for stalemate
export const stalemateBoard = () => {
  const squares = emptyBoard();
  squares[1] = squarePiece(new King(1));
  squares[6] = squarePiece(new Rook(1));
  squares[21] = squarePiece(new Knight(1));
  squares[39] = squarePiece(new King(0));
  squares[62] = squarePiece(new Queen(1));
  return squares;
};

export const stalemateBoard2 = () => {
  const squares = stalemateBoard();
  squares[44] = squares[62];
  squares[62] = null;
  return squares;
};

export const checkBoard = () => {
  const squares = emptyBoard();
  squares[0] = squarePiece(new Bishop(1));
  squares[9] = squarePiece(new King(0));
  return squares;
};

//move queen 62 -> 44 for checkmate
export const checkmateBoard = () => {
  const squares = emptyBoard();
  squares[1] = squarePiece(new King(1));
  squares[6] = squarePiece(new Rook(1));
  squares[11] = squarePiece(new Bishop(1));
  squares[21] = squarePiece(new Knight(1));
  squares[39] = squarePiece(new King(0));
  squares[62] = squarePiece(new Queen(1));
  return squares;
};

export const checkmateBoard2 = () => {
  const squares = checkmateBoard();
  squares[42] = squares[62];
  squares[62] = null;
  return squares;
};

export const checkmateBoard3 = () => {
  const squares = checkmateBoard2();
  squares[32] = squarePiece(new Rook(1));
  return squares;
};

export const players = (p1, p2) => [
  {
    graveyard: [],
    graveyardPick: false,
    name: p1,
    moves: []
  },
  {
    graveyard: [],
    graveyardPick: false,
    name: p2,
    moves: []
  }
];

const initialState = (player1 = "", player2 = "") => {
  return {
    board: initialBoard(),
    moves: 0,
    players: [
      {
        graveyard: [],
        graveyardPick: false,
        name: player1,
        moves: []
      },
      {
        graveyard: [],
        graveyardPick: false,
        name: player2,
        moves: []
      }
    ],
    kingAttacked: [false, false],
    checkmate: -1,
    invalidMove: false,
    stalemate: false
  };
};

export const partialState = (player1 = "", player2 = "") => {
  return {
    board: partialBoard(),
    moves: 0,
    players: [
      {
        graveyard: [],
        graveyardPick: false,
        name: player1,
        moves: []
      },
      {
        graveyard: [],
        graveyardPick: false,
        name: player2,
        moves: []
      }
    ],
    kingAttacked: [false, false],
    checkmate: -1,
    invalidMove: false,
    stalemate: false
  };
};

export function unserialize(name, player) {
  var piece;
  switch (name) {
    case pieces.pawn.name:
      piece = new Pawn(player);
      break;
    case pieces.bishop.name:
      piece = new Bishop(player);
      break;
    case pieces.rook.name:
      piece = new Rook(player);
      break;
    case pieces.knight.name:
      piece = new Knight(player);
      break;
    case pieces.queen.name:
      piece = new Queen(player);
      break;
    default:
      piece = new King(player);
      break;
  }
  return piece;
}

export default initialState;
