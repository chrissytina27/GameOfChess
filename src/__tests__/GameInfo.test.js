import GameInfo from "../components/GameInfo";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import { players } from "../helpers/initialBoard";

describe("GameInfo", () => {
  const testProps = {
    playerTurn: 1,
    graveyardPick: {},
    endGame: jest.fn(),
    resetGame: jest.fn(),
    players: players("p1", "p2"),
    check: false,
    checkmate: false,
    invalidMove: false,
    staleMate: false
  };

  it("renders correctly", () => {
    const gameInfo = GameInfo(testProps);
    const tree = renderer.create(gameInfo).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
