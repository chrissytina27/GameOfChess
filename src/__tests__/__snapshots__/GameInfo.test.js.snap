// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameInfo renders correctly 1`] = `
<div
  className="nav-wrapper box2"
>
  <div
    className="nav1"
  >
    <div
      className="icon"
    >
      <img
        alt="White Player Logo"
        className="player-icon1"
        src={Object {}}
      />
    </div>
  </div>
  <div
    className="nav2"
  >
    <span
      className="dot"
    />
  </div>
  <div
    className="nav3"
  >
    <div
      className="check"
    >
      White , please select a piece from the graveyard to replace your pawn.
    </div>
    <div
      className="check"
    >
      
    </div>
    <div
      className="check"
    >
      
    </div>
    <div
      className="check"
    >
      
    </div>
    <div
      className="table-box"
    >
      <table>
        <thead>
          <tr>
            <th>
              White
            </th>
            <th>
              Black
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>
  <div
    className="nav6"
  >
    <div
      className="icon-dot"
    >
      <img
        alt="Black Player Logo"
        className="player-icon2"
        src={Object {}}
      />
    </div>
  </div>
  <div
    className="nav7"
  >
     
    <span />
  </div>
  <div
    className="nav8"
  >
    <button
      onClick={[MockFunction]}
    >
      End Game
    </button>
    <button
      onClick={[MockFunction]}
    >
      Reset Game
    </button>
  </div>
</div>
`;
